### PR TITLE
#59 ITERATORS AND COMPARATORS COMMIT

### PR DESCRIPTION
**Pull request by ivelin980**

### CODE REVIEW

File: javaAdvanced_iteratorsAndComparators/src/exercise/comparingobjects/Person.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	private String town;
+	
+	public Person(String name, int age, String town) {
+		this.name = name;
+		this.age = age;
+		this.town = town;
+	}
+	
+	@Override
+	public int compareTo(Person p) {
```
Comment (ngavrailov) [inline javaAdvanced_iteratorsAndComparators/src/exercise/comparingobjects/Person.java:15]: formating

File: javaAdvanced_iteratorsAndComparators/src/exercise/stackiterator/Node.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package exercise.stackiterator;
+
+public class Node {
```
Comment (ngavrailov) [inline javaAdvanced_iteratorsAndComparators/src/exercise/stackiterator/Node.java:3]: This class is not used.

File: javaAdvanced_iteratorsAndComparators/src/exercise/strategypattern/ComparatorByName.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package exercise.strategypattern;
+
+import java.util.Comparator;
+
+public class ComparatorByName implements Comparator<Person> {
+
+	@Override
+	public int compare(Person first, Person second) {
+		int result = first.getName().length() - second.getName().length();
```
Comment (ngavrailov) [inline javaAdvanced_iteratorsAndComparators/src/exercise/strategypattern/ComparatorByName.java:9]: Maybe it is good to have checks for null. If first is null or first.getName() is null.

File: javaAdvanced_iteratorsAndComparators/src/lab/book/Main.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	public static void main(String[] args) {
+		Book bookOne = new Book("Animal Farm", 2003, "George Orwell");
+		Book bookThree = new Book("The Documents in the Case", 2002);
+		Book bookTwo = new Book("The Documents in the Case", 1930, "Dorothy Sayers", "Robert Eustace");
+
+		List<Book> books = new ArrayList<>();
+		books.add(bookOne);
+		books.add(bookTwo);
+		books.add(bookThree);
+		for (Book book : books) {
```
Comment (ngavrailov) [inline javaAdvanced_iteratorsAndComparators/src/lab/book/Main.java:28]: This print is not working. You didn't implement toString() in Book.

Comment (ngavrailov) [reply javaAdvanced_iteratorsAndComparators/src/lab/book/Main.java:28]: Auto generated toString() in Book and System.out.println(books); will works fine

File: javaAdvanced_iteratorsAndComparators/src/lab/book/comparator/BookComparator.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package lab.book.comparator;
+
+import java.util.Comparator;
+
+public class BookComparator implements Comparator<Book> {
+
+	@Override
+	public int compare(Book bookOne, Book bookTwo) {
+		if(bookOne.getTitle().compareTo(bookTwo.getTitle()) == 0) {
+			return Integer.compare(bookOne.getYear(), bookTwo.getYear());
```
Comment (ngavrailov) [inline javaAdvanced_iteratorsAndComparators/src/lab/book/comparator/BookComparator.java:10]: Formating.

File: javaAdvanced_iteratorsAndComparators/src/lab/library/Library.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package lab.library;
+
+import java.util.Iterator;
+
+public class Library implements Iterable<Book>{
+	private Book[] books;
+	
+	public Library(Book...books) {
```
Comment (ngavrailov) [inline javaAdvanced_iteratorsAndComparators/src/lab/library/Library.java:8]: Formating

**Review by ngavrailov — CHANGES_REQUESTED** [review]

File: javaAdvanced_iteratorsAndComparators/src/lab/book/Main.java (hunk 4 above)
Comment (ivelin980) [reply javaAdvanced_iteratorsAndComparators/src/lab/book/Main.java:28]: fixed

File: javaAdvanced_iteratorsAndComparators/src/exercise/strategypattern/ComparatorByName.java (hunk 3 above)
Comment (ivelin980) [reply javaAdvanced_iteratorsAndComparators/src/exercise/strategypattern/ComparatorByName.java:12]: fixed

File: javaAdvanced_iteratorsAndComparators/src/exercise/stackiterator/Node.java (hunk 2 above)
Comment (ivelin980) [reply javaAdvanced_iteratorsAndComparators/src/exercise/stackiterator/Node.java]: fixed

File: javaAdvanced_iteratorsAndComparators/src/exercise/comparingobjects/Person.java (hunk 1 above)
Comment (ivelin980) [reply javaAdvanced_iteratorsAndComparators/src/exercise/comparingobjects/Person.java:15]: fixed

File: javaAdvanced_iteratorsAndComparators/src/lab/book/comparator/BookComparator.java (hunk 5 above)
Comment (ivelin980) [reply javaAdvanced_iteratorsAndComparators/src/lab/book/comparator/BookComparator.java:10]: fixed

File: javaAdvanced_iteratorsAndComparators/src/lab/library/Library.java (hunk 6 above)
Comment (ivelin980) [reply javaAdvanced_iteratorsAndComparators/src/lab/library/Library.java]: fixed

**Review by ngavrailov — APPROVED** [review]

**Pull request merged**